discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -178,9 +155,16 @@ def query_ipfs_metadata_json(self, gateway_ipfs_urls, default_metadata_fields):
                         default_metadata_fields, r.json()
                     )
                     # Exit loop if dict is successfully retrieved
-                    logger.warning(f"IPFSCLIENT | Retrieved from {url}")
+                    logger.info(
+                        f"IPFSCLIENT | query_ipfs_metadata_json Retrieved from {url} took {time.time() - start_time} seconds"
+                    )
+                    start_shutdown = time.time()
                     self.force_clear_queue_and_stop_task_execution(executor)
-                    break
+                    logger.info(
+                        f"IPFSCLIENT | shutdown queue took {time.time() - start_shutdown} seconds"
+                    )
+                    return formatted_json
+
                 except Exception as exc:
                     logger.error(f"IPFSClient | {url} generated an exception: {exc}")
         return formatted_json
```

Comment:
let's just return None here to avoid confusion

Reply:
yup